sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/implementation/AccessTokensImpl.java

```diff
@@ -0,0 +1,243 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.containers.containerregistry.implementation;
+
+import com.azure.containers.containerregistry.implementation.models.AccessToken;
+import com.azure.containers.containerregistry.implementation.models.AcrErrorsException;
+import com.azure.containers.containerregistry.implementation.models.PathsV3R3RxOauth2TokenPostRequestbodyContentApplicationXWwwFormUrlencodedSchema;
+import com.azure.core.annotation.BodyParam;
+import com.azure.core.annotation.ExpectedResponses;
+import com.azure.core.annotation.Get;
+import com.azure.core.annotation.HeaderParam;
+import com.azure.core.annotation.Host;
+import com.azure.core.annotation.HostParam;
+import com.azure.core.annotation.Post;
+import com.azure.core.annotation.QueryParam;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceInterface;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.annotation.UnexpectedResponseExceptionType;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.RestProxy;
+import com.azure.core.util.Context;
+import com.azure.core.util.FluxUtil;
+import reactor.core.publisher.Mono;
+
+/** An instance of this class provides access to all the operations defined in AccessTokens. */
+public final class AccessTokensImpl {
+    /** The proxy service used to perform REST calls. */
+    private final AccessTokensService service;
+
+    /** The service client containing this operation class. */
+    private final ContainerRegistryImpl client;
+
+    /**
+     * Initializes an instance of AccessTokensImpl.
+     *
+     * @param client the instance of the service client containing this operation class.
+     */
+    AccessTokensImpl(ContainerRegistryImpl client) {
+        this.service =
+                RestProxy.create(AccessTokensService.class, client.getHttpPipeline(), client.getSerializerAdapter());
+        this.client = client;
+    }
+
+    /**
+     * The interface defining all the services for ContainerRegistryAccessTokens to be used by the proxy service to
+     * perform REST calls.
+     */
+    @Host("{endpoint}")
+    @ServiceInterface(name = "ContainerRegistryAcc")
+    private interface AccessTokensService {
```

Comment:
To prevent the issue in #17368 you'll want to add `service-interface-as-public: true` to the code generator configurations to generate this as a public interface.